### PR TITLE
feat: hive docker 

### PR DESCRIPTION
**Pull request by Eikix**
Time spent on this PR:

Resolves: #670 

Context:
Hive test suite needs to run inside a docker

telegram-cloud-photo-size-4-5918109500030697628-y

# Pull Request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build-related changes
- [ ] Documentation content changes
- [ ] Testing

- - -
This change is [Reviewable](https://reviewable.io/reviews/kkrt-labs/kakarot-rpc/764)

### CODE REVIEW

File: docker/hive/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+FROM quay.io/apibara/sink-mongo as indexer
+FROM debian:bookworm as apibara-build
+RUN apt-get update && apt-get install -y patchelf && rm -rf /var/lib/apt/lists/*
+# Run `docker image inspect apibara/starknet:1.2.0` to get the exact path
+COPY --from=apibara /nix/store/swzxd9vbhhj92qzh16j48vb011q57v59-apibara-starknet-1.2.0/bin/apibara-starknet /usr/local/bin/starknet
+COPY --from=indexer /nix/store/3iqnrcirqpg4s7zdy1wdh0dq17jwzmlc-apibara-sink-mongo-0.5.3/bin/apibara-sink-mongo /usr/local/bin/sink-mongo
+# Change the interpreter path.
+# x86-64  "/lib64/ld-linux-x86-64.so.2"
+# aarch64 "/lib/ld-linux-aarch64.so.1"
+RUN patchelf --set-interpreter /lib/ld-linux-aarch64.so.1 /usr/local/bin/starknet && \
```
Comment (greged93) [inline docker/hive/Dockerfile:18]: This probably needs to be improved in order to update the interpreter path based on the arch...

File: docker/hive/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+ENV APIBARA_AUTH_TOKEN=""
+ENV MONGO_CONNECTION_STRING=mongodb://localhost:27017
+ENV MONGO_DATABASE_NAME=kakarot-local
+ENV STARTING_BLOCK=0
+ENV STREAM_URL=http://localhost:7171
+ENV SINK_TYPE=mongo
+### Kakarot RPC environment variables
+### Port 8545: https://github.com/ethereum/hive/blob/master/simulators/ethereum/rpc/helper.go#L50
+ENV KATANA_ACCOUNT_ADDRESS=0x01b44e32c8edbcada35fa7ae6f1f44bbc16657f442b7b19286812e1976c34d83
+ENV KATANA_PRIVATE_KEY=0x05b8e27acc53020ed612507af8245cbffafd8aa588b79decaee1acf99a1c5cc2
```
Comment (greged93) [inline docker/hive/Dockerfile:52]: Used for automatic deployment of EOA for hive

File: docker/hive/Dockerfile (hunk 1 above)
Comment (Eikix) [reply docker/hive/Dockerfile:18]: why do we hard code here something like to arm?

File: docker/hive/Dockerfile (hunk 2 above)
Comment (Eikix) [reply docker/hive/Dockerfile:52]: Do we need to be careful that newer versions of katana don't have different prefunded accts?

File: docker/hive/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,100 @@
+# trunk-ignore-all(terrascan/AC_DOCKER_0047)
+
+# CairoVM Chain
+FROM ghcr.io/dojoengine/dojo:v0.5.1 as katana
```
Comment (Eikix) [inline docker/hive/Dockerfile:4]: do we need to use v0.6.0-alpha?

Comment (Eikix) [reply docker/hive/Dockerfile:4]: [v0.6.0-alpha.0](https://github.com/dojoengine/dojo/releases/tag/v0.6.0-alpha.0)

Comment (greged93) [reply docker/hive/Dockerfile:4]: I prefer to stay on 0.5.1 for now and bump in another PR

File: docker/hive/Dockerfile (hunk 2 above)
Comment (greged93) [reply docker/hive/Dockerfile:52]: We will need to change this maybe yes

**Review by Eikix — COMMENTED** [review]
let's add this docker build in the ci on release?

**Review by greged93 — APPROVED** [review]

**Pull request merged**